lib/helpers.js

```diff
@@ -117,3 +117,68 @@ function outputBuffer (value) {
   if (Buffer.isBuffer(value)) return value
   return buffer(value.slice(2), 'hex')
 }
+
+
+function constructFilterRelationQuery( typeMap,
+                                       recordTypes,
+                                       typesPathToParent,
+                                       pathSegmentsToParent,
+                                       whereOperation,
+                                       value,
+                                       query = '' ){
+
+  if( !typesPathToParent.length ){
+    return query
+  }
+
+  const pathSegment = pathSegmentsToParent[0]
+  const currentType = typesPathToParent[0]
+  if(!query){
+    //TODO: can I always assume id here?
+    query = `SELECT id FROM "${typeMap[currentType] || currentType}" WHERE ${whereOperation(pathSegment, value)} \n`
+  }
+  else {
+    const recordType = recordTypes[currentType]
+    if(!recordType[pathSegment].isArray) {
+      query = `SELECT id FROM "${typeMap[currentType] || currentType}" WHERE  "${pathSegment}" IN ( ${query} ) \n`
+    }
+    else{
+      query = `SELECT id FROM "${typeMap[currentType] || currentType}" WHERE  "${pathSegment}" && ARRAY( ${query} ) \n`
```

Comment:
i think this can be refactored so the `isArray` check is inline in the string template.

```js
`... ${isArray ? '&& ARRAY' : 'IN'} ...`
```

Reply:
should be fixed